### PR TITLE
PLNSRVCE-730: [M9]Migrate infra credentials to Bitwarden

### PR DESCRIPTION
**Pull request by bnallapeta**
- Fetch secrets from Bitwarden and replace those values in the corresponding files.
- Docs directory updated with relevant examples.
- SRE directory updated with README.
- Enable/disable tracing to hide secrets.

Local run output (truncated):

```
  - Deploy compute:
    [Bitwarden]:   
      bitwarden.yaml file exists. Checking if the required variables to connect to Bitwarden are set.
      Required variables are available.
      Extracted secrets from Bitwarden and substituted them in the relevant files.
    [Compute a9265aeffb3cc4f94a9f7138a8ef389e-ffc1c735f0a0b5bc-elb-us-east-1-amazonaws-com]:
    - Installing shared manifests... 
        namespace/tekton-results configured
        secret/tekton-results-database unchanged
    - Installing applications via Openshift GitOps... 
        application.argoproj.io/pipeline-service unchanged
    - Checking deployment status
        - tekton-pipelines-controller: Exists, Ready

```

### CODE REVIEW

File: operator/docs/sre/examples/single_cluster/credentials/secrets/bitwarden.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+---
```
Comment (Roming22) [inline operator/docs/sre/examples/single_cluster/credentials/secrets/bitwarden.yaml:1]: Remove the bitwarden example from single_cluster to keep easy to understand.

File: operator/gitops/sre/credentials/secrets/README.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
+# Secrets from secret managers
+
+This directory contains files that store secrets from secret managers like Bitwarden, Vault, AWS Secret manager etc.
+An example of such a secret could be credentials (username and password) to connect to a remote database such as AWS RDS.
+
+The format in which the secrets need to be available in this directory is shown below:
```
Comment (Roming22) [inline operator/gitops/sre/credentials/secrets/README.md:6]: You need specify that the secret is the content of the file encoded in base64.

File: operator/images/cluster-setup/bin/install.sh [hunk 3]
```diff
@@ -108,6 +118,68 @@ switch_cluster() {
...
+
+    if ! mkdir -p "$(dirname "$WORKSPACE_DIR/$path")" 2>/dev/null; then
+      printf "Unable to create the folder. Exiting.\n" >&2 | indent 2
+      exit 1
+    fi
+    if ! echo "$content" | base64 -d > "$WORKSPACE_DIR/$path"; then
+      printf "Unable to copy the contents of the secret to the specified path. Exiting.\n" >&2 | indent 2
+      exit 1
+    fi
+    setx_on
```
Comment (Roming22) [inline operator/images/cluster-setup/bin/install.sh:178]: Add a printf statement list the file and id.

File: operator/images/cluster-setup/bin/install.sh [hunk 4]
```diff
@@ -76,6 +76,16 @@ parse_args() {
   done
 }
 
+# turns off tracing even with set -x mode enabled across the script
```
Comment (Roming22) [inline operator/images/cluster-setup/bin/install.sh:79]: `to prevent secrets leaking`

File: operator/images/cluster-setup/bin/install.sh (hunk 3 above)
Comment (bnallapeta) [reply operator/images/cluster-setup/bin/install.sh:178]: Done.

**Review by Roming22 — APPROVED** [review]

**Pull request merged**